app/api/routes/chat.py

```diff
@@ -56,8 +56,23 @@ def query(chat_request: ChatRequest):
 
 @router.post("/learn", status_code=204)
 def learn(learn_request: LearnRequest):
-    learn_collection.add(
-        documents=[learn_request.text],
-        ids=[uuid.uuid4().hex],
-    )
+
+    documents_paragraphs = learn_request.text.split("\n")
+
+    for document_sequence in documents_paragraphs:
+
+        split_into = 4
+        quarter = len(document_sequence) // split_into
+        document_quarters = [
+            document_sequence[i * quarter : (i + 1) * quarter]
+            for i in range(split_into - 1)
+        ]
+        document_quarters.append(document_sequence[(split_into - 1) * quarter :])
+
+        for document in document_quarters:
+            learn_collection.add(
+                documents=[document],
```

Comment:
Since the argument accepts multiple documents at the same time, it might be better at performance to rely on its internal functionality instead of using the for-loop ourselves. I haven't read the code of ChromaDB, but maybe they are doing a batch operation to efficiently insert multiple records in database

I would consider doing something like this

```
learn_collection.add(
  documents=document_quarters,
  ids=[uuid.uuid4().hex for _ in range(SPLIT_INTO)],
)
```

Reply:
Sounds, good!  I will feed them at the same time